File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {

        return f(0, nums.length - 1, nums, target);
        
    }

    int f(int l, int r, int[] nums, int t){
        if(l>r) return -1;
        int m = (l+r)/2;
        if(nums[m] == t) return m;
        else if(nums[m] > t) return f(l, m-1, nums, t);
        else if(nums[m] < t) return f(m+1, r, nums, t);

        return -1;
    }


}

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-0.cpp
class Solution {
    private:
    struct cmp {
        // true , then b moves closer to top
        bool operator()(int a, int b){
            return a > b;
        }
    };


    // kth largest - 5th largest
    // use min heap - of size limit ok k - 5 - ans is top
    priority_queue<int, vector<int>, cmp> pq;

public:
    int findKthLargest(vector<int>& nums, int k) {
        for(auto& n : nums){
            pq.push(n);
            if(pq.size() == k+1) pq.pop();
        }    

    return pq.top();
    }
};
